fast_response/listeners/gw_gcn_listener.py

```diff
@@ -40,37 +43,71 @@ def process_gcn(payload, root):
               for elem in root.iterfind('.//Param')}
 
     # Read trigger time of event
-    isot = lxml.etree.tostring(root.find('./WhereWhen/ObsDataLocation/ObservationLocation/'
-                               'AstroCoords/Time/TimeInstant/ISOTime'))
+    eventtime = root.find('.//ISOTime').text
+    event_mjd = Time(eventtime, format='isot').mjd
 
-    # Parse time string
-    time_string = isot.split('>')[1].split('<')[0]
-    date, time = time_string.split('T')
-    trigger = date + ' ' + time
+    print('GW trigger time: %s \n' % Time(eventtime, format='isot').iso)
 
-    print('\n')
-    print('GW trigger time: %s \n' % trigger)
+    current_mjd = Time(datetime.utcnow(), scale='utc').mjd
+    needed_delay = 1000./84600./2.
+    current_delay = current_mjd - event_mjd
+    while current_delay < needed_delay:
+        print("Need to wait another {:.1f} seconds before running".format(
+            (needed_delay - current_delay)*86400.)
+            )
+        time.sleep((needed_delay - current_delay)*86400.)
+        current_mjd = Time(datetime.utcnow(), scale='utc').mjd
+        current_delay = current_mjd - event_mjd
 
     skymap = params['skymap_fits']
     name = root.attrib['ivorn'].split('#')[1]
-    name = 'GW190517\_055101'
-    skymap = '/data/user/rhussain/fitsFiles/corrected_GWTC2/GW190517_055101_PublicationSamples.hdf5'
+    name= name.split('-')[0] #remove? this removes the map identifiers from the name
 
-    #Print all parameters.
-    for key, value in params.items():
-        print(key, '=', value)
+    if root.attrib['role']=='test':
```

Comment:
Is this actually included in the XML files?

Reply:
from what I see, it looks like it's usually root.attrib['role']='observation' in the xml files, meaning that it is coming from an actual event. Raamis seems to have used this as a check to see if people needed to be called/notified. I set it to 'test' so that it would fail that check, but am definitely open to changing this

Reply:
I don't think there will ever be a time that `root['role']=='test'`, unless you assign it by hand before time, so it might make more sens to do `if root['role'] != 'observation'`?

Reply:
Fixed in https://github.com/icecube/FastResponseAnalysis/pull/24/commits/cc053cc7feebeec862207d55509ea6aa4875764c